lecture_3/03_project/calculator/internal/engine/handler/handler.go

```diff
@@ -0,0 +1,98 @@
+package handler
+
+import (
+	"context"
+	"github.com/mattn/go-sqlite3"
+	domainmodels "github.com/superbet-group/code-cadets-2021/lecture_3/03_project/calculator/internal/domain/models"
+	rabbitmqmodels "github.com/superbet-group/code-cadets-2021/lecture_3/03_project/calculator/internal/infrastructure/rabbitmq/models"
+	"log"
+)
+
+type Handler struct {
+	betRepository CalcBetRepository
+}
+
+func New(betRepository CalcBetRepository) *Handler {
+	return &Handler{
+		betRepository: betRepository,
+	}
+}
+
+func (h *Handler) HandleBets(ctx context.Context, bets <-chan rabbitmqmodels.Bet ) {
+	go func() {
+		for bet := range bets {
+			log.Println("Processing bet, betId:", bet.Id, " with selectionId:", bet.SelectionId)
+
+			domainBet := domainmodels.Bet{
+				Id:                   bet.Id,
+				SelectionId:          bet.SelectionId,
+				SelectionCoefficient: bet.SelectionCoefficient,
+				Payment:              bet.Payment,
+			}
+
+			err := h.betRepository.InsertCalcBet(ctx, domainBet)
+			if err == sqlite3.ErrConstraint {
+				log.Println("bet already exists in calc_bets")
+				continue
+			} else if err != nil {
+				log.Println("Failed to insert bet, error: ", err)
+				continue
+			}
```

Comment:
This handler should not be aware that `sqlite3` is used under the hood. That information should be hidden and exposed differently.

E.g. you could change the signature of your `repository` method to:
`InsertCalcBet(ctx context.Context, bet domainmodels.Bet) (bool, error)`.
And then, through `bool` return value you can signal if it was actually inserted or not.